lib/charms/layer/apache_zeppelin.py

```diff
@@ -168,3 +240,44 @@ def close_ports(self):
     def cleanup(self):
         self.dist_config.remove_dirs()
         unitdata.kv().set('zeppelin.installed', False)
+
+
+class ZeppelinAPI(object):
+    """
+    Helper for interacting with the Appache Zeppelin REST API.
+    """
+    def _url(self, *parts):
+        dc = utils.DistConfig()
+        url = 'http://localhost:{}/api/'.format(dc.port('zeppelin'))
+        for part in parts:
+            url = urljoin(url, part)
+        return url
+
+    def import_notebook(self, contents):
+        response = requests.post(self._url('notebook'), data=contents)
+        if response.status_code != 201:
+            return None
+        return response.json()['body']
+
+    def delete_notebook(self, notebook_id):
+        requests.delete(self._url('notebook/', notebook_id))
+
+    def modify_interpreter(self, interpreter_name, properties):
```

Comment:
Danger!  this requires zeppelin to be started.  In the reactive handlers, you are guarding calls to this function by `zeppelin.started`, but that state isn't always true.  For example, if you restart the unit post deployment, zeppelin will not auto-start, yet the `zeppelin.started` state will be set from the initial deployment.

We either need to jps test for zeppelin in this method (and not call requests.get if zepp is not started), or jps test from the reactive handler (starting zepp if necessary) before calling this method.